### ISSUE TITLE
Find a better way to keep haxe server running when closing/opening windows

### ISSUE THREAD

**Issue by kevinresol** [snowkit/atom-haxe#59]
As titled, atom-haxe tries to start `haxe --wait` for each atom window which results in `Error: Couldn't wait on 127.0.0.1:6112` (can be seen when toggled completion debug log), if there is already a haxe server started by another atom window. 

This is not a big deal, because the completion still works. It is just wasting a little CPU.

**Comment by jeremyfa** [comment]
Yes, that's actually a temporary workaround to prevent the server from not being restarted when working with multiple windows and closing the one that has started the server.

Without this workaround, you could kill the server without knowing it when closing the window (and wonder why completion does not work anymore).

That said we should for sure look for a better way of handling this.

**Comment by kevinresol** [comment]
arghhh, I see. That make sense.

**Comment by kevinresol** [comment]
Just a small suggestion, what if check only when a completion is needed?

**Comment by jeremyfa** [comment]
Yes, that could be an option. Will have to see that after the [haxe rewrite](https://github.com/snowkit/atom-haxe/tree/haxe-rewrite) is done.
